internal/workflow/activities/bundle.go

```diff
@@ -24,6 +26,7 @@ type BundleActivityParams struct {
 	Key              string
 	TempFile         string
 	StripTopLevelDir bool
+	BatchDir         string
 }
```

Comment:
I can't think of better approaches to deal with the complexity in this activity but having some tests covering all cases would be super useful.